src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -89,40 +91,80 @@ public void update()
             {
                 if (i == index)
                 {
-                    if (buildingEntry.getValue() instanceof AbstractBuildingWorker
-                          && !list.contains(buildingEntry.getValue())
-                          && ((AbstractBuildingWorker) buildingEntry.getValue()).needsAnything())
+                    if(buildingEntry.getValue() instanceof AbstractBuildingWorker
+                            && !list.contains(buildingEntry.getValue())
+                            && ((AbstractBuildingWorker) buildingEntry.getValue()).needsAnything())
                     {
                         checkInWareHouse((AbstractBuildingWorker) buildingEntry.getValue(), true);
                     }
+                    else if(buildingEntry.getValue() instanceof BuildingHome && ((BuildingHome) buildingEntry.getValue()).isFoodNeeded())
+                    {
+                        checkInWareHouseForFood((BuildingHome) buildingEntry.getValue(), true);
+                    }
                     this.index++;
                 }
                 i++;
             }
         }
     }
 
+    public boolean checkInWareHouseForFood(final BuildingHome buildingEntry, final boolean addToList)
+    {
+        if (buildingEntry.isFoodNeeded())
+        {
+            if (isInHut(itemStack -> !InventoryUtils.isItemStackEmpty(itemStack) && itemStack.getItem() instanceof ItemFood))
+            {
+                if (addToList)
+                {
+                    buildingEntry.setOnGoingDelivery(true);
+                    list.add(buildingEntry);
+                }
+                return true;
+            }
+
+            if (list.contains(buildingEntry))
+            {
+                list.remove(buildingEntry);
+                buildingEntry.setOnGoingDelivery(false);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get the first task in the list.
+     * @return the building which needs a delivery.
+     */
+    @Nullable
+    public AbstractBuilding getTask()
+    {
+        if(list.isEmpty())
+        {
+            return null;
+        }
+        return list.remove(0);
+    }
+
     /**
      * Check if the required items by the building are in the wareHouse.
-     *
      * @param buildingEntry the building requesting.
-     * @param addToList     if is in warehouse should add to the list?
+     * @param addToList if is in warehouse should add to the list?
      * @return true if has something in warehouse to deliver.
      */
     public boolean checkInWareHouse(@NotNull final AbstractBuilding buildingEntry, boolean addToList)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)